File: order/src/main/java/com/trinhvu/order/viewmodel/order/OrderVm.java
package com.trinhvu.order.viewmodel.order;

import com.trinhvu.order.model.Order;
import com.trinhvu.order.model.enumeration.OrderStatus;
import com.trinhvu.order.model.enumeration.PaymentStatus;

import java.math.BigDecimal;
import java.util.Set;
import java.util.stream.Collectors;

public record OrderVm(
        Long id,
        String email,
        String note,
        int numberItem,
        BigDecimal totalPrice,
        OrderStatus orderStatus,
        PaymentStatus paymentStatus,
        Long paymentId,
        Set<OrderItemVm> orderItemVms
) {
    public static OrderVm fromModel(Order order) {
        Set<OrderItemVm> orderItemVms = order.getOrderItems().stream()
                .map(OrderItemVm::fromModel)
                .collect(Collectors.toSet());

        return new OrderVm(
                order.getId(),
                order.getEmail(),
                order.getNote(),
                order.getNumberItem(),
                order.getTotalPrice(),
                order.getOrderStatus(),
                order.getPaymentStatus(),
                order.getPaymentId(),
                orderItemVms
        );
    }
}

File: stock-watch-fe/components/order/models/OrderVm.ts
import { OrderItemVm } from "./OrderItemVm";

export interface OrderVm {
  id?: number;
  email: string;
  note: string;
  numberItem: number;
  totalPrice: number; // BigDecimal represented as number in TypeScript
  orderStatus: string;
  paymentStatus: string;
  paymentId?: number;
  orderItemVms: OrderItemVm[]; // Set to manage the unique list of items

  checkoutId: string;
}
